### PR TITLE
Parallelize tests on circle.

### PR DESCRIPTION
**Pull request by chriseth**
- [x] split into build and test

This runs commandline tests and soltest in parallel on the same machine (it has 2CPUs).

### CODE REVIEW

File: scripts/tests.sh [hunk 1]
```diff
@@ -39,60 +39,65 @@ then
     fi
     testargs_no_opt="--logger=JUNIT,test_suite,$2/no_opt.xml"
     testargs_opt="--logger=JUNIT,test_suite,$2/opt.xml"
+    shift
+    shift
```
Comment (axic) [inline scripts/tests.sh]: Why do you need to remove argv's here?

Comment (chriseth) [reply scripts/tests.sh]: Another try that was since removed did some further processing of following arguments.

File: scripts/tests.sh [hunk 2]
```diff
@@ -39,60 +39,65 @@ then
...
-# Wait until the IPC endpoint is available.  That won't be available instantly.
-# The node needs to get a little way into its startup sequence before the IPC
-# is available and is ready for the unit-tests to start talking to it.
-while [ ! -S /tmp/test/geth.ipc ]; do sleep 2; done
-echo "--> IPC available."
-sleep 2
-# And then run the Solidity unit-tests (once without optimization, once with),
-# pointing to that IPC endpoint.
 echo "--> Running tests without optimizer..."
-  "$REPO_ROOT"/build/test/soltest --show-progress $testargs_no_opt -- --ipcpath /tmp/test/geth.ipc && \
```
Comment (axic) [inline scripts/tests.sh:91]: I liked `--show-porgess`, any reason removing it?

File: scripts/tests.sh [hunk 3]
```diff
@@ -39,60 +39,65 @@ then
...
-  echo "--> Running tests WITH optimizer..." && \
-  "$REPO_ROOT"/build/test/soltest --show-progress $testargs_opt -- --optimize --ipcpath /tmp/test/geth.ipc
-ERROR_CODE=$?
+"$REPO_ROOT"/build/test/soltest $testargs_no_opt -- --ipcpath /tmp/test/geth.ipc && \
+echo "--> Running tests WITH optimizer..." && \
+"$REPO_ROOT"/build/test/soltest $testargs_opt -- --optimize --ipcpath /tmp/test/geth.ipc &
+SOLTEST_PID=$!
```
Comment (axic) [inline scripts/tests.sh]: Doesn't these here ignore the exit code `soltest`?

File: scripts/tests.sh (hunk 2 above)
Comment (chriseth) [reply scripts/tests.sh:91]: The output of the commandline tests and the end-to-end tests is interleaved now, so it did not make too much sense. I can add an option to not interleave them or only interleave them if `$CI` is set.

File: scripts/tests.sh (hunk 3 above)
Comment (chriseth) [reply scripts/tests.sh]: we use `set -e`

**Comment by chriseth** [comment]
Updated.

File: scripts/tests.sh (hunk 3 above)
Comment (axic) [reply scripts/tests.sh]: But it is a subshell, does it apply there too?

Comment (chriseth) [reply scripts/tests.sh]: Why is it a subshell?

Comment (axic) [reply scripts/tests.sh]: Sorry not subshell, but the fork (it had `&`) before.

After your change it is not parallel anymore and isn't affected by this question.

**Comment by chriseth** [comment]
@axic approved?

**Review by axic — APPROVED** [review]

**Pull request merged**